tests/rest/client/utils.py

```diff
@@ -145,7 +146,7 @@ def create_room_as(
 
     def invite(
         self,
-        room: Optional[str] = None,
+        room: str,
```

Comment:
It doesn't make sense to invite to, leave, or ban from an unspecified room. Do you know why this annotation allowed an `Optional` room in the first place? Are/were there any tests taking advantage of this?

Reply:
It calls `change_membership`.
There it is required and also needed for the path.

https://github.com/matrix-org/synapse/blob/2b0f1575a9145cabba92beea8969af2390d147eb/tests/rest/client/utils.py#L252-L283

Reply:
It was part of the inital Homeserver commit:
https://github.com/matrix-org/synapse/commit/4f475c7697722e946e39e42f38f3dd03a95d8765#diff-7d9372b9840620c86d8739fc6be631da0bcdcc4b41101b435dc4429ad34b6094